alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php

```diff
@@ -252,7 +279,46 @@ public static function file_put_contents ( FileSyncKey $key , $content , $strict
 		kFile::filePutContents($fullPath , $content);
 		self::setPermissions($fullPath);
 		self::createSyncFileForKey($rootPath, $filePath,  $key , $strict , !is_null($res), false, md5($content), kPathManager::getStorageProfileIdForKey($key));
-		self::encryptByFileSyncKey($key, kPathManager::getStorageProfileIdForKey($key));
+		$encryptedContent = self::encryptByFileSyncKey($key, kPathManager::getStorageProfileIdForKey($key));
+		$contentToPut = $encryptedContent ? $encryptedContent : $content;
+		$compressedContentToPut = gzcompress($contentToPut);
+		$contentToPut = $compressedContentToPut ? $compressedContentToPut : $contentToPut;
+
+		//If the file is under a specific threshold in size, save it to redis cache
+		$contentSize = strlen($contentToPut);
+		$fileSyncFilesToCache = kConf::get(self::FILE_SYNC_TYPES_TO_CACHE, kConfMapNames::RUNTIME_CONFIG, array());
+		$maxFileSizeToCache = kConf::get(self::MAX_FILE_SIZE_TO_CACHE, kConfMapNames::RUNTIME_CONFIG);
```

Comment:
If MAX_FILE_SIZE_TO_CACHE is not set, need to return after attempting to get it from kConf